src/errors.rs

```diff
@@ -4,6 +4,7 @@ use std::fmt;
 
 /// The kind of an error (non-exhaustive)
 #[derive(Debug)]
+#[non_exhaustive]
```

Comment:
Technically a breaking change. I would remove that change just in case.

Reply:
It was non exhaustive before. I don't see why it's a breaking change

Reply:
It was faking non-exhaustiveness, you could still match on `__Nonexhaustive`. With the new `#[non_exhaustive]` you can't, and the attribute is not there anymore so it might break some weird `match` people have. Never underestimate the weird code that can happen!

Reply:
I reverted the manual_non_exhaustive change, and added an exception for clippy with reasoning, why we kept our implementation.